src/plugins/data/common/search/aggs/metrics/metric_agg_type.ts

```diff
@@ -67,9 +79,13 @@ export class MetricAggType<TMetricAggConfig extends AggConfig = IMetricAggConfig
 
         // Return proper values when no buckets are present
         // `Count` handles empty sets properly
-        if (!bucket[agg.id] && isSettableToZero) return 0;
+        if (!bucket[agg.id] && isSettableToZero && !agg.params.emptyAsNull) return 0;
```

Comment:
does this make isSettableToZero flag obsolete ? should we make it possible to set every metric to zero when emptyAsNull is not set ?

Reply:
`isSettableToZero` is not really a flag, it's a hardcoded lookup in this `getValue` implementation. It's there to make sure there's _always_ a zero even if Elasticsearch is not sending one which is kind of the opposite thing as `emptyAsNull`. I'm not 100% sure in which cases this happens, from what I can tell Elasticsearch will always report back the 0 in the same way. I just added the condition here to make sure it's not changing things in some weird edge case.